base/event.jl

```diff
@@ -105,12 +105,39 @@ function wait()
             end
         else
             t = shift!(Workqueue)
-            t.state == :queued || throw(AssertionError("shift!(Workqueue).state == :queued"))
+            if t.state != :queued
+                # assume this somehow got queued twice,
+                # probably broken now, but try discarding this switch and keep going
+                # can't throw here, because it's probably not the fault of the caller to wait
+                # and don't want to use print() here, because that may try to incur a task switch
+                ccall(:jl_safe_printf, Void, (Ptr{UInt8}, Vararg{Int32}),
+                    "\nWARNING: Workqueue inconsistency detected: shift!(Workqueue).state != :queued\n")
+                continue
+            end
             arg = t.result
             t.result = nothing
             t.state = :runnable
-            result = yieldto(t, arg)
-            current_task().state == :runnable || throw(AssertionError("current_task().state == :runnable"))
+            local result
+            try
```

Comment:
then we should make `try` cheaper. possible cost is never a good argument for skipping error handling.

it protects the system against `yieldto` throwing an error (for example, if we got here because a finalizer tried to use `yield()`, the state of the containing task and the task we tried to schedule would be corrupted afterwards, through no fault of their own, and henceforth be unusable from the scheduler. if one of those tasks was the REPL, you would generally then be rapidly and unceremoniously).